### PR TITLE
[docs] Release notes for 2.20.2.0

### PR DESCRIPTION
**Pull request by aishwarya24**
To do - Add highlights if needed
YB- https://deploy-preview-21262--infallible-bardeen-164bc9.netlify.app/preview/releases/release-notes/v2.20/#new-features
YBA - https://deploy-preview-21262--infallible-bardeen-164bc9.netlify.app/preview/releases/yba-releases/v2.20/#new-features

### CODE REVIEW

File: docs/content/preview/releases/release-notes/v2.20.md [hunk 1]
```diff
@@ -58,6 +58,165 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+docker pull yugabytedb/yugabyte:2.20.2.0-b145
+```
+
+### New features
+
+* Added support for read-only mode for DR replica, currently supported for YSQL tables. {{<badge/ga>}}
+
+* Tablet splitting now can be enabled on tables with CDC configured. {{<badge/ga>}} {{<issue 18479>}}
+
+* Transactional CDC now supports consistent snapshots, ensuring data integrity during replication to a sink. Snapshots, ordered by commit time across all tables and tablets, establish a reliable replication order. {{<badge/ga>}}, {{<issue 19682>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/release-notes/v2.20.md:102]:
```suggestion
* Transactional CDC now supports consistent snapshots, ensuring data integrity during replication to a sink. Snapshots, ordered by commit time across all tables and tablets, establish a reliable replication order. {{<badge/ga>}} {{<issue 19682>}}
```

File: docs/content/preview/releases/release-notes/v2.20.md [hunk 2]
```diff
@@ -58,6 +58,165 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+* Enhances CDCSDK to communicate tablet splits to clients as soon as they are detected during `GetChanges` calls, offering improved data handling via hash_key checks and facilitating modifications to tests. {{<issue 18479>}}
+* Adds yb-admin CLI support for creating consistent snapshot streams with options NOEXPORT_SNAPSHOT and USE_SNAPSHOT, guarded by the TEST flag yb_enable_cdc_consistent_snapshot_streams. {{<issue 19682>}}
+* Sets the CDC Consistent Snapshot feature as a preview with `USE_SNAPSHOT` as the default consistent snapshot option while creating a stream, and guards changes by the PREVIEW flag `yb_enable_cdc_consistent_snapshot_streams`. {{<issue 20367>}} -->
+
+### Improvements
+
+#### YSQL
+
+* Introduces yb_silence_advisory_locks_not_supported_error as a temporary solution, avoiding disruption while users transition from the use of advisory locks. {{<issue 19974>}}
+* Shifts from the test gflag "FLAGS_TEST_enable_db_catalog_version_mode" to the tech preview gflag "FLAGS_ysql_enable_db_catalog_version_mode", enhancing user control over concurrent DDL execution across different databases. {{<issue 12417>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/release-notes/v2.20.md:121]: tech preview? badge?

File: docs/content/preview/releases/release-notes/v2.20.md [hunk 3]
```diff
@@ -58,6 +58,165 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+* Issues a notice for unsafe ALTER TABLE operations, including for ADD COLUMN...DEFAULT, to indicate existing rows won't be backfilled with the default value, enhancing user awareness. Suppression possible by setting ysql_suppress_unsafe_alter_notice gflag to true. {{<issue 19360>}}
+* Added sorting capabilities to BatchNestedLoopJoin to return the rows in the same order as NestedLoopJoin {{<issue 19589>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/release-notes/v2.20.md:123]:
```suggestion
* Added sorting capabilities to BatchNestedLoopJoin to return the rows in the same order as NestedLoopJoin. {{<issue 19589>}}
```

File: docs/content/preview/releases/release-notes/v2.20.md [hunk 4]
```diff
@@ -58,6 +58,165 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+* Replaces the `ysql_max_read_restart_attempts` and `ysql_max_write_restart_attempts` gflags with `yb_max_query_layer_retries`, applies limit to Read Committed isolation statement retries, and adjusts `retry_backoff_multiplier` and `retry_min_backoff` defaults. {{<issue 20359>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/release-notes/v2.20.md:124]: here the flags are in code - why not everywhere?

File: docs/content/preview/releases/release-notes/v2.20.md [hunk 5]
```diff
@@ -58,6 +58,165 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+* Introduces two new boolean flags for YSQL webserver debugging and failure identification.The first will log every endpoint access and its list of arguments before running the path handler. The second will print basic tcmalloc stats after the path handler and garbage collection have run. {{<issue 20157>}}
+* Restarts postmaster in critical sections.This applies to backend PG code; defines and improves handling of errors in these sections. {{<issue 20255>}}
+* Introduces the `pg_stat_statements.yb_qtext_size_limit` flag to control the maximum file size read into memory, preventing system crashes due to oversized or corrupt qtext files. {{<issue 20211>}}
+* Caps the number of attempts to read inconsistent backend entries to 1000 for safer operation and better visibility, limiting potential infinite loops in /rpcz calls and logging every hundredth attempt. {{<issue 20274>}}
+* Resolves segmentation faults in the webserver SIGHUP handler during cleanup. {{<issue 20309>}}
+* Display consistent wait-start times in `pg_locks`  view for waiting transactions {{<issue 18603>}},{{<issue 20120>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/release-notes/v2.20.md:167]:
```suggestion
* Display consistent wait-start times in `pg_locks`  view for waiting transactions. {{<issue 18603>}},{{<issue 20120>}}
```

File: docs/content/preview/releases/release-notes/v2.20.md [hunk 6]
```diff
@@ -58,6 +58,165 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+* Reduces excessive memory consumption during secondary index scans {{<issue 20275>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/release-notes/v2.20.md:168]:
```suggestion
* Reduces excessive memory consumption during secondary index scans. {{<issue 20275>}}
```

File: docs/content/preview/releases/release-notes/v2.20.md [hunk 7]
```diff
@@ -58,6 +58,165 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+* Allows the finalize_plan function to now apply finalize_primnode on PushdownExprs, ensuring Parameter nodes in subplans transfer values accurately, especially in Parallel Queries. {{<issue 19694>}}
+* Return correct results when Batch Nested Loop join is used for queries involving Nested LEFT JOINs on LATERAL views {{<issue 19642>}},{{<issue 19946>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/release-notes/v2.20.md:170]:
```suggestion
* Return correct results when Batch Nested Loop join is used for queries involving Nested LEFT JOINs on LATERAL views. {{<issue 19642>}},{{<issue 19946>}}
```

File: docs/content/preview/releases/release-notes/v2.20.md [hunk 8]
```diff
@@ -58,6 +58,165 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+* Upgrades "Unknown session" error to FATAL permitting drivers to instantly terminate stale connections, minimizing manual user intervention. {{<issue 16445>}}
+* Rectifies correctness issues when BatchNestedLoop join is used and the join condition contained a mix of equality and non-equality filters. {{<issue 20531>}}
+* Mitigates incorrect data entry into the default partition by incrementing the schema version when creating a new partition, enhancing data consistency across all connections. {{<issue 17942>}}
+* Rectifies correctness issue when join on inequality condition and join columns contains NULL values {{<issue 20642>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/release-notes/v2.20.md:174]:
```suggestion
* Rectifies correctness issue when join on inequality condition and join columns contains NULL values. {{<issue 20642>}}
```

File: docs/content/preview/releases/release-notes/v2.20.md [hunk 9]
```diff
@@ -58,6 +58,165 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+* Prevents "Not enough live tablet servers to create table" error during ALTER TABLE SET TABLESPACE by correctly supplying the placement_uuid, even when creating a table in the same tablespace. {{<issue 14984>}}
+* Addresses a bug that caused backup failure due to the absence of `yb_catalog_version`, by ensuring the function's existence post-normal migration. {{<issue 18507>}}
+* Corrects an error in the aggregate scans' pushdown eligibility criteria to prevent wrong results from being returned when PG recheck is not expected, but YB preliminary check is required to filter additional rows. {{<issue 20709>}}
+* Ensures the Linux `PDEATH_SIG` mechanism signals child processes of their parent process's exit, by correctly configuring all PG backends immediately after their fork from the postmaster process. {{<issue 20396>}}
+* Fixes a MISMATCHED_SCHEMA error when upgrading from version 2.16 to 2.21 by introducing a 2-second delay for catalog version propagation when a breaking DDL statement is detected. {{<issue 20842>}}
+* Return correct result for join queries using BatchNestedLoop  join and DISTINCT scan using the range Index during inner table scan  {{<issue 20827>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/release-notes/v2.20.md:182]:
```suggestion
* Return correct result for join queries using BatchNestedLoop  join and DISTINCT scan using the range Index during inner table scan. {{<issue 20827>}}
```

File: docs/content/preview/releases/release-notes/v2.20.md [hunk 10]
```diff
@@ -58,6 +58,165 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+* Renders a fix for memory corruption issue that caused failure in creating a valid execution plan for `SELECT DISTINCT` queries. Enables successful execution of queries without errors and prevents server connection closures by disabling `distinct pushdown`. This fix improves the stability and effectiveness of SELECT DISTINCT queries. {{<issue 20893>}}
+* Eliminates unnecessary computation of range bounds in Index-Only Scan precheck condition, preventing crashes for certain queries {{<issue 21004>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/release-notes/v2.20.md:184]:
```suggestion
* Eliminates unnecessary computation of range bounds in Index-Only Scan precheck condition, preventing crashes for certain queries. {{<issue 21004>}}
```

File: docs/content/preview/releases/yba-releases/v2.20.md [hunk 11]
```diff
@@ -53,6 +53,191 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+ </li>
+</ul>
+
+### New features
+
+<!-- * Exposes the `replicated-migrate` flag as an {{<badge/ea>}} feature with modified help text, making commands accessible when `YBA_MODE` is set to `EA`, improving the user interface of scheduled backups along with button visibility. PLAT-11274,PLAT-11096 -->
+
+* [Nutanix Ready](../../../integrations/nutanix-ahv/). YugabyteDB and YugabyteDB Anywhere has been certified to be compatible as part of the Nutanix Ready AHV program.
+
+* Automated [Linux OS patching](../../../yugabyte-platform/manage-deployments/upgrade-nodes/) and upgrades for on-prem. You can now utilize safety-improved YugabyteDB  Anywhere APIs when automating you rolling patch scripts for Linux OSs (typically, for OS security, and minor or major OS upgrades). PLAT-10591, PLAT-10988
```
Comment (ddhodge) [inline docs/content/preview/releases/yba-releases/v2.20.md:79]:
```suggestion
* Automated [Linux OS patching](../../../yugabyte-platform/manage-deployments/upgrade-nodes/) and upgrades for on-prem. You can now use safety-improved YugabyteDB  Anywhere APIs when automating your rolling patch scripts for Linux OSs (typically, for OS security, and minor or major OS upgrades). PLAT-10591, PLAT-10988
```

File: docs/content/preview/releases/yba-releases/v2.20.md [hunk 12]
```diff
@@ -53,6 +53,191 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+
+* YugabyteDB Anywhere (YBA) and YugabyteDB version compatibility list. You can now independently upgrade YBA without upgrading the YugabyteDB universes. All supported YugabyteDB versions are compatible. PLAT-11260
+
+* Improved Auth when using [GCS for backups for Kubernetes universes](../../../yugabyte-platform/back-up-restore-universes/configure-backup-storage/#google-cloud-storage). When taking backups of a Kubernetes universe to Google GCP, allow use of the Google GKE's Service Account for authentication to the Google Storage. PLAT-10571
+
+* Asymmetric nodes in each region (API support only). In a stretched cluster, nodes in each region can now have different instance types (that is, different number of CPU cores, memory, and so on).
+
+* [Task Retryability](/stable/yugabyte-platform/manage-deployments/retry-failed-task/) for VM-based universes. You can now address the cause of a failure for a complex operation fails and retry. For example, if adding nodes to a universe fails ( due to a cloud quota limit), you can now address the cloud issue and retry. {{<badge/ea>}} PLAT-6141
```
Comment (ddhodge) [inline docs/content/preview/releases/yba-releases/v2.20.md:89]:
```suggestion
* [Task Retryability](/stable/yugabyte-platform/manage-deployments/retry-failed-task/) for VM-based universes. You can now address the cause of a failure for a complex operation and retry. For example, if adding nodes to a universe fails due to a cloud quota limit, you can fix the quota issue and retry the task. {{<badge/ea>}} PLAT-6141
```

File: docs/content/preview/releases/yba-releases/v2.20.md [hunk 13]
```diff
@@ -53,6 +53,191 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+
+* [Backup portability](../../../yugabyte-platform/back-up-restore-universes/restore-universe-data/#advanced-restore-procedure). After sending backups to a cloud object store (For example, S3, GCS, Azure blob storage), you can now copy or transfer those backup sets to other cloud storage buckets and still restore successfully. {{<badge/ea>}} PLAT-10179
```
Comment (ddhodge) [inline docs/content/preview/releases/yba-releases/v2.20.md:91]:
```suggestion
* [Backup portability](../../../yugabyte-platform/back-up-restore-universes/restore-universe-data/#advanced-restore-procedure). After sending backups to a cloud object store (such as S3, GCS, or Azure blob storage), you can now copy those backup sets to other cloud storage buckets and restore from them. {{<badge/ea>}} PLAT-10179
```

File: docs/content/preview/releases/yba-releases/v2.20.md [hunk 14]
```diff
@@ -53,6 +53,191 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+
+* [xCluster DR](../../../yugabyte-platform/back-up-restore-universes/disaster-recovery/). A turnkey solution for Business Continuity and Disaster Recovery to recover from an unplanned outage and planned maintenance. Replication occurs asynchronously (using [xCluster Replication](../../../architecture/docdb-replication/async-replication/)) from one universe to another. {{<badge/ea>}}
```
Comment (ddhodge) [inline docs/content/preview/releases/yba-releases/v2.20.md:93]:
```suggestion
* [xCluster DR](../../../yugabyte-platform/back-up-restore-universes/disaster-recovery/). A turnkey solution for business continuity and disaster recovery, allowing you to recover from an unplanned outage (failover) or to perform a planned switchover. {{<badge/ea>}}
```

File: docs/content/preview/releases/yba-releases/v2.20.md [hunk 15]
```diff
@@ -53,6 +53,191 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+
+* [Edit universe's provider](../../../yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/gcp/): add/remove regions, change service account. Expanded support for reconfiguring a universe. You can now add or remove an availability zone/region/or an AMI, and modify CSP credentials when a universe's provider is actively in-use.
```
Comment (ddhodge) [inline docs/content/preview/releases/yba-releases/v2.20.md:95]:
```suggestion
* [Edit universe providers](../../../yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/gcp/). You can now add and remove regions and change service accounts for providers with active universes.
```

File: docs/content/preview/releases/yba-releases/v2.20.md [hunk 16]
```diff
@@ -53,6 +53,191 @@ For more information, see [#20827](https://github.com/yugabyte/yugabyte-db/issue
...
+
+<!-- * Exposes the `replicated-migrate` flag as an {{<badge/ea>}} feature with modified help text, making commands accessible when `YBA_MODE` is set to `EA`, improving the user interface of scheduled backups along with button visibility. PLAT-11274,PLAT-11096 -->
+
+* [Nutanix Ready](../../../integrations/nutanix-ahv/). YugabyteDB and YugabyteDB Anywhere has been certified to be compatible as part of the Nutanix Ready AHV program.
+
+* Automated [Linux OS patching](../../../yugabyte-platform/manage-deployments/upgrade-nodes/) and upgrades for on-prem. You can now utilize safety-improved YugabyteDB  Anywhere APIs when automating you rolling patch scripts for Linux OSs (typically, for OS security, and minor or major OS upgrades). PLAT-10591, PLAT-10988
+
+* REST API versioning and clean-up. REST API simplification and streamlining for most important APIs. PLAT-11133
+
+* YugabyteDB Anywhere (YBA) and YugabyteDB version compatibility list. You can now independently upgrade YBA without upgrading the YugabyteDB universes. All supported YugabyteDB versions are compatible. PLAT-11260
```
Comment (ddhodge) [inline docs/content/preview/releases/yba-releases/v2.20.md:83]: True?
Ticket number doesn't look related

File: docs/content/preview/releases/release-notes/v2.20.md (hunk 4 above)
Comment (aishwarya24) [reply docs/content/preview/releases/release-notes/v2.20.md:124]: Because this gflag is not documented at all.
